modules/cli/pkg/runs/runsUpdate.go

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package runs
+
+import (
+	"slices"
+	"context"
+	"log"
+	"net/http"
+
+	"github.com/galasa-dev/cli/pkg/api"
+	"github.com/galasa-dev/cli/pkg/embedded"
+	galasaErrors "github.com/galasa-dev/cli/pkg/errors"
+	"github.com/galasa-dev/cli/pkg/galasaapi"
+	"github.com/galasa-dev/cli/pkg/spi"
+)
+
+// ---------------------------------------------------
+
+// RunsUpdate - performs all the logic to implement the `galasactl runs update` command,
+// but in a unit-testable manner.
+func RunsUpdate(
+	runName string,
+	addTags []string,
+	removeTags []string,
+	console spi.Console,
+	commsClient api.APICommsClient,
+	timeService spi.TimeService,
+) error {
+	var err error
+
+	log.Printf("RunsUpdate entered.")
+
+	if runName == "" {
+		return galasaErrors.NewGalasaError(galasaErrors.GALASA_ERROR_MISSING_NAME_FLAG, "--name")
+	}
+
+	if len(addTags) == 0 && len(removeTags) == 0 {
+		err = galasaErrors.NewGalasaError(galasaErrors.GALASA_ERROR_UPDATE_RUN_MISSING_FIELD, "--add-tags or --remove-tags")
+		return err
+	}
+
+	// Validate the runName as best we can without contacting the ecosystem.
+	err = ValidateRunName(runName)
+	if err != nil {
+		return err
+	}
+
+	addTags = removeDuplicateTags(addTags)
+	removeTags = removeDuplicateTags(removeTags)
+
+	// Check for tags that are both added and removed.
+	for _, tag := range addTags {
+		if (slices.Contains(removeTags, tag)) {
+			return galasaErrors.NewGalasaError(galasaErrors.GALASA_ERROR_UPDATE_RUN_INVALID_TAG_UPDATE, tag)
+		}
+	}
```

Comment:
Would it be worth moving this validation into its own function and then calling the validation function here?